### PR TITLE
review: feat: read jvm version for default compliance level

### PR DESCRIPTION
**Pull request by MartinWitt**
Maybe the current JVM version is a better default compared to 8 or?

### CODE REVIEW

**Comment by SirYwell** [comment]
I think that's indeed a better default, but it's probably something we want to at least highlight in the next release? I also don't think we need to catch any exception around the version code, as it should work on Java 10 onwards and we're targeting Java 17.

**Comment by MartinWitt** [comment]
> I think that's indeed a better default, but it's probably something we want to at least highlight in the next release?

Hmm, we could start collecting common answers and important behaviors on the website.

**Comment by SirYwell** [comment]
> Hmm, we could start collecting common answers and important behaviors on the website.

Good idea. But maybe we can also improve the in-code documentation (e.g. add a comment to the `DEFAULT_CODE_COMPLIANCE_LEVEL` field). WDYT?

File: src/main/java/spoon/support/StandardEnvironment.java [hunk 1]
```diff
@@ -65,7 +65,15 @@ public class StandardEnvironment implements Serializable, Environment {
 
 	private static final long serialVersionUID = 1L;
 
-	public static final int DEFAULT_CODE_COMPLIANCE_LEVEL = 8;
+	/**
+	 *
+	 * Only features available in the compliance level are correctly parsed by spoon.
+	 * So if you use 11, spoon cant parse records. By default, we use the language level of the executing jvm.
+	 * If you want to parse java 21 code with a java 17 jvm you need set the compliance level with {@link  #setComplianceLevel}
+	 * to least 21.
```
Comment (SirYwell) [inline src/main/java/spoon/support/StandardEnvironment.java:73]:
```suggestion
	 * Only features available in the compliance level are correctly parsed by spoon.
	 * By default, spoon uses the language level of the executing JVM. So if you use Java 11, spoon can't parse records.
	 * If you want to parse Java 21 code with a Java 17 JVM you need set the compliance level with {@link  #setComplianceLevel}
	 * to at least 21.
```

**Review by SirYwell — APPROVED** [review]
:+1:

**Pull request merged**